File: spaniel/src/models/schema/checked-ideas.ts
export type CheckedKeywordDto = {
  keyword: string;
  generatorVersions: number[];
  tokenlists: string[];
  runsCount: number;
};

export type CheckedKeyword = CheckedKeywordDto;

export type CheckedPasslistDto = {
  passlistUrl: string;
  name: string;
};

export type CheckedPasslist = CheckedPasslistDto;

export type CheckedIdeasDto = {
  checkedKeywords: CheckedKeyword[];
  checkedPasslists: CheckedPasslist[];
};

export type CheckedIdeas = CheckedIdeasDto;

export const mapCheckedIdeas = (dto: CheckedIdeasDto) => ({
  ...dto,
});

File: shepherd/internal/common/models/checked_ideas.go
package models

type UsedKeyword struct {
	Keyword           string                      `bson:"keyword" json:"keyword"`
	GeneratorVersions []TokenGeneratorVersionEnum `bson:"generatorVersions" json:"generatorVersions"`
	Tokenlists        []string                    `bson:"tokenlists" json:"tokenlists"`
	RunsCount         int                         `bson:"runsCount" json:"runsCount"`
}

type UsedPasslist struct {
	PasslistUrl string `bson:"passlistUrl" json:"passlistUrl"`
	Name        string `bson:"name" json:"name"`
}

type UsedIdeas struct {
	CheckedKeywords  []*UsedKeyword  `json:"checkedKeywords"`
	CheckedPasslists []*UsedPasslist `json:"checkedPasslists"`
}
